### DISCUSSION TITLE
Feature Request: Read key

### DISCUSSION THREAD

**Discussion by CommanderB** [ybizeul/StreamDeckWS#19] [Ideas]
Would it possible to read a key contents (title, image, state) without a trigger event?

State is a good example, that is only available on a key press/appear. But unless that is written to a global variable there isnt a way to know what state the key is in. Title is similar, if dynamic text is being written, it needs to be held elsewhere before it can be acted upon without a trigger. 

Thanks for listening!

**Comment by ybizeul** [comment]
Looks like I missed that message. Replying for posterity.

If I understand your question, I think it's already possible by catching the `willAppear` messages ? Now I can't remember exactly what is put in that message.

**Comment by CommanderB** [comment]
Thank you! I too have forgotten - I have worked through what I needed. As I write a new key title, I save that to a global variable, then on next press I read the global. Its working. 

Thanks again for a great plugin. Its still in hourly use as part of my daily routines.
